ngraph/core/src/tensor.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/tensor.hpp"
+
+#include "blob_factory.hpp"
+#include "ie_ngraph_utils.hpp"
+
+
+namespace ov {
+
+#define TENSOR_STATEMENT(...)                                           \
+    OV_ASSERT(_impl != nullptr, "Tensor was not initialized.");         \
+    try {                                                               \
+        __VA_ARGS__;                                                    \
+    } catch (...) {                                                     \
+        ::InferenceEngine::details::Rethrow();                          \
+    }
+
+
+Tensor::Tensor(const std::shared_ptr<void>& so,
+               const std::shared_ptr<ie::Blob>& impl)
+    : _so{so},
+      _impl{impl} {
+    if (_impl == nullptr) {
+        IE_THROW() << "Tensor was not initialized.";
+    }
+}
+
+Tensor::Tensor(const element::Type element_type, const Shape& shape) {
+    _impl = make_blob_with_precision({ie::details::convertPrecision(element_type), {shape.begin(), shape.end()}, ie::Layout::NCHW});
```

Comment:
can we just omit information about layout?

Reply:
done